src/GeositeFramework/plugins/layer_selector_v2/WmsService.js

```diff
@@ -0,0 +1,139 @@
+define([
+        "dojo/_base/declare",
+        "dojo/Deferred",
+        "underscore",
+        "./lib/xmlToJson",
+        "framework/util/ajax",
+        "./util",
+        "./LayerNode",
+    ],
+    function(declare,
+             Deferred,
+             _,
+             xmlToJson,
+             ajaxUtil,
+             util,
+             LayerNode) {
+        "use strict";
+
+        return declare(null, {
+            constructor: function(server) {
+                this.server = server;
+            },
+
+            getServiceUrl: function() {
+                return util.urljoin(this._getBaseServiceUrl() + '?request=GetCapabilities&service=WMS');
+            },
+
+            _getBaseServiceUrl: function() {
+                // Likely an AGS WMS service
+                if (this.server.name) {
+                    return util.urljoin(this.server.url, this.server.name, 'MapServer/WMSServer');
+                } else {
+                    return this.server.url;
+                }
+            },
+
+            // Return a promise containing map service data.
+            fetchMapService: function() {
+                return ajaxUtil.fetch(this.getServiceUrl(), 'text', this._transformServiceData);
+            },
+
+            // Return cached map service data.
+            getServiceData: function() {
+                return ajaxUtil.get(this.getServiceUrl());
+            },
+
+            // Return a promise with service layer data.
+            fetchLayerDetails: function(tree, layerId) {
+                return new Deferred().resolve(function() {
+                    return tree.findLayer(layerId);
+                });
```

Comment:
These should probably just return null or an empty object since there are no layer details for a WMS layer.